symbolic-symcache/src/writer.rs

```diff
@@ -381,8 +381,16 @@ where
             return Ok(*index);
         }
 
-        if self.files.len() >= std::u16::MAX as usize {
-            return Err(SymCacheErrorKind::TooManyValues(ValueKind::File).into());
+        // Instead of failing hard when exceeding the maximum allowed number of files (u16::MAX),
+        // we rather insert a sentinel "<unknown>" file and use that for all files that exceed the max.
+        if self.files.len() > u16::MAX as usize {
```

Comment:
in this version the final size of `self.files.len()` will be `u16::MAX + 1` will it not?  Does it not have to stay inside `u16`?

Reply:
⬆️  This version will create `SymCacheErrorKind::TooManyValues` when writing the files segment. Also for safety, please ensure that `self.files` never contains more than `u16::MAX` entries. This shows intent more clearly and is more robust.

Reply:
I’m surprised how difficult it is to think through. Lets consider u8 since thats easier. `MAX` is 255 and `.len()` is 256, because of 0-based indexing, `MAX` is the largest *index* that you can use.